src/Microsoft.VisualStudio.Composition.AppHost/CreateComposition.cs

```diff
@@ -18,81 +24,476 @@ namespace Microsoft.VisualStudio.Composition.AppHost
+                        assembly = Assembly.Load(name);
+                        return true;
+                    }
+                    catch (BadImageFormatException)
+                    {
+                        // This happens for some reference assemblies that can only be loaded in a reflection-only context.
+                        // But we're not allowed in an assembly resolve event to return reflection-only loaded assemblies.
+                        // So just return false to communicate the failure.
+                    }
+                }
+            }
+
+            assembly = null;
+            return false;
+        }
+
+        private static IEnumerable<string> GetCatalogErrorLines(ComposableCatalog catalog)
+        {
+            foreach (var error in catalog.DiscoveredParts.DiscoveryErrors)
+            {
+                yield return error.GetUserMessage();
+
+                if (error.InnerException is ReflectionTypeLoadException reflectionTypeLoadException)
+                {
+                    // The LoaderExceptions have a tendency to have a lot of redundancy.
+                    // So get just the unique set to report.
+                    foreach (string uniqueMessage in reflectionTypeLoadException.LoaderExceptions.Select(e => e.Message).Distinct())
+                    {
+                        yield return "\t" + uniqueMessage;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetCompositionErrorLines(CompositionConfiguration composition)
+        {
+            int level = 1;
+            foreach (var errors in composition.CompositionErrors)
+            {
+                yield return $"******************* Composition error level {level++} ********************";
+                foreach (var error in errors)
+                {
+                    yield return error.Message;
+                }
+
+                yield return string.Empty;
+            }
+        }
+
+        private void LogLines(string filePath, IEnumerable<string> lines, CancellationToken cancellationToken)
+        {
+            Requires.NotNullOrEmpty(filePath, nameof(filePath));
+            Requires.NotNull(lines, nameof(lines));
+
+            cancellationToken.ThrowIfCancellationRequested();
+            try
+            {
+                using (var writer = new StreamWriter(new FileStream(filePath, FileMode.Create, FileAccess.Write, FileShare.Read)))
+                {
+                    foreach (string line in lines)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        writer.WriteLine(line);
+                    }
+
+                    writer.Flush();
+                }
+
+                this.writtenFiles.Add(filePath);
+            }
+            catch (IOException ex)
+            {
+                this.Log.LogWarning("Unable to write log file: \"{0}\". {1}", filePath, ex.Message);
```

Comment:
No. I saw this warning once, and I'd like to understand it more. If it's an overbuild/multiproc build bug, it should be fixed rather than ignored. If it's something else, I want folks to complain so I can understand it and fix it myself. :)

Reply:
I added a code comment explaining this.